GitUI/GitUICommands.cs

```diff
@@ -537,12 +537,8 @@ bool Action()
                     // if the dialog is loaded on its own, plugins need to be loaded before we load the form
                     if (!werePluginsRegistered)
                     {
-                        ThreadHelper.JoinableTaskFactory.Run(async () =>
-                        {
-                            PluginRegistry.Initialize();
-                            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                            PluginRegistry.Register(this);
-                        });
+                        PluginRegistry.Initialize();
+                        PluginRegistry.Register(this);
```

Comment:
Is this switching to the UI thread inside?

Reply:
No, the UI thread must be executing already

Reply:
> Is this switching to the UI thread inside?

This function creates and shows FormCommit right afterwards. So it should be the UI thread.
Should I add `ThrowIfNotOnUIThread` here?

> We shouldn't be initialising plugins on the main thread though...

The main thread is blocked intentionally here if FormCommit is opened as stand-alone:

```
                    // Load plugins synchronously
                    // if the commit dialog is opened from the main form, all plugins are already loaded and we return instantly,
                    // if the dialog is loaded on its own, plugins need to be loaded before we load the form
                    if (!werePluginsRegistered)
```

Reply:
> Should I add ThrowIfNotOnUIThread here?

I think so, and add a comment

In FormBrowse plugins are loaded async:

```c#
ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
            {
                await TaskScheduler.Default;
                PluginRegistry.Initialize();
                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                RegisterPlugins();
            }).FileAndForget();
```

Reply:
Thank you. I should've read the comment above (I wouldn't be surprised if I was the author, heh). It looks odd, but thinking of it, it's probably justified - when we're opening the main form (i.e., FormBrowse) we don't expect any immediate interactions with plugins. However, in the commit dialog the plugin interactions were believed to be more likely (like JIRA commit plugin?). 

I suppose it's valid to question that assumption, and change it, if necessary.